Server/SQL/procedures.sql

```diff
@@ -263,5 +307,35 @@ BEGIN
     ) AS DT ON DT.tag_id=Tag.id;
 END$$
 
+CREATE PROCEDURE SEARCH_DECKS(IN query_string VARCHAR(255), IN tags VARCHAR(255), IN num_tags Integer)
+BEGIN
+
+    SELECT
+        Deck.uuid,
+        Deck.name,
+        Deck.rating,
+        (SELECT COUNT(*) FROM Rating WHERE deck_id=Deck.uuid) AS num_ratings,
+        Deck.owner,
+        Deck.public,
+        Deck.version AS deck_version,
+        L.version AS user_data_version,
+        Deck.created,
+        Deck.last_update,
+        L.last_update_device,
+        Deck.share_code
+    FROM Deck LEFT JOIN (
+        SELECT version, last_update_device, deck_id
+        FROM Library
+    ) AS L ON L.deck_id=Deck.uuid
+    WHERE Deck.public=TRUE
+    AND (LOWER(Deck.name) like CONCAT('%',LOWER(query_string),'%')
+        OR Deck.uuid IN (SELECT dt.deck_id FROM DeckTag dt
+	    LEFT JOIN Tag t ON dt.tag_id=t.id
+	    WHERE FIND_IN_SET(t.tag,tags)
```

Comment:
Well find_in_set isn't case insensitive right? So we would have to make things lowercase first... So instead of adding that we can just force all tags to be lowercase like you said.